py-polars/polars/api.py

```diff
@@ -43,7 +43,6 @@ def __get__(self, instance: NS | None, cls: type[NS]) -> NS | type[NS]:
             return self._ns
 
         ns_instance = self._ns(instance)  # type: ignore[call-arg]
-        setattr(instance, self._accessor, ns_instance)
```

Comment:
I think this were not required even before, but it served as "cache" on instance of expression.
Now it gives error because instance does not have __dict__ and _accessor goes straight to descriptor - and it's read only.

But since expressions are frequently recreated, then this "cache" were only working on chained namespace access like
```pl.col("foo").namespace_foo.func().namespace_foo.other_func()```

Reply:
I'd appreciate if @alexander-beedie could sign off on this change before this gets merged. I don't really see exactly what is going on here.

Reply:
Turns out this did have user impact:
https://github.com/pola-rs/polars/issues/14851

For this reason, and some related issues popping up, I will be reverting this PR.